### PR TITLE
Kube integration

### PR DESCRIPTION
**Pull request by rajatchopra**
@mrunalp PTAL

### CODE REVIEW

**Comment by mrunalp** [comment]
We need to replace jq with python.

File: ovs_subnet/bin/openshift-ovs-subnet [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    nsenter -n -t $pid -- $add_subnet_route
+
+    add_default_cmd="ip route add default via $tap_ip"
+    nsenter -n -t $pid -- $add_default_cmd
+}
+
+Teardown() {
+    veth_host=$(jq .network_state.veth_host /var/lib/docker/execdriver/native/${net_container}/state.json | tr -d '"')
+    ovs_port=$(ovs-ofctl -O OpenFlow13 dump-ports-desc br0  | grep ${veth_host} | cut -d "(" -f 1 | tr -d ' ')
+    ovs-vsctl del-port $veth_host
```
Comment (mrunalp) [inline ovs_subnet/bin/openshift-ovs-subnet]: || true

File: ovs_subnet/kube_controller/controller.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
+package kube_controller
```
Comment (mrunalp) [inline ovs_subnet/kube_controller/controller.go]: Need to take out _ from package name.

File: ovs_subnet/kube_controller/controller.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+func (c *FlowController) manageLocalIpam(ipnet *net.IPNet) error {
+	ipamHost := "127.0.0.1"
+	ipamPort := uint(9080)
+	inuse := make([]string, 0)
+	ipam, _ := netutils.NewIPAllocator(ipnet.String(), inuse)
+	f, err := os.Create("/etc/openshift-sdn/config.env")
+	if err != nil {
+		return err
+	}
+	if err != nil {
```
Comment (mrunalp) [inline ovs_subnet/kube_controller/controller.go]: Take out this block.

File: ovs_subnet/common.go [hunk 4]
```diff
@@ -1,27 +1,18 @@
-package controller
+package ovs_subnet
```
Comment (mrunalp) [inline ovs_subnet/common.go]: Rename to remove _

**Comment by mrunalp** [comment]
@rajatchopra Build failed with gofmt errors.

**Comment by mrunalp** [comment]
LGTM

**Comment by jeremyeder** [comment]
@mrunalp When will this land in openshift?

**Comment by mrunalp** [comment]
@jeremyeder Hopefully by the end of this sprint. API integration work is going on right now.

**Pull request merged**